### ISSUE TITLE
 'NoneType' object has no attribute 'text'

### ISSUE THREAD

**Issue by cerinawithasea** [digitalec/sonus#12]
I had to predownload the files because the covers are screwing up the script today, but i had a minute so i thought i could share this'
I got this (i will attach a link to get the files and the rest)
```
sonus /Users/cerinawithasea/Music/Kali\ Wallace\ -\ Shallow\ Graves
Output will be saved to: /Users/cerinawithasea/Music
Found 9 files, containing 49 chapters...
Extracting chapter Shallow Graves from Part01.mp3
Extracting chapter Chapter 1 from Part01.mp3
Extracting chapter Chapter 2 from Part01.mp3
Extracting chapter Chapter 3 from Part01.mp3
Extracting chapter Chapter 4 from Part01.mp3
Extracting chapter Chapter 5 from Part01.mp3
Extracting chapter Chapter 6 from Part01.mp3
Extracting chapter Chapter 7 from Part01.mp3
Extracting chapter Chapter 8 from Part01.mp3
Extracting chapter Chapter 9 from Part01.mp3
Extracting chapter Chapter 10 from Part02.mp3
Extracting chapter Chapter 11 from Part02.mp3
Extracting chapter Chapter 12 from Part02.mp3
Extracting chapter Chapter 13 from Part02.mp3
Extracting chapter Chapter 14 from Part02.mp3
Extracting chapter Chapter 15 from Part03.mp3
Extracting chapter Chapter 16 from Part03.mp3
Extracting chapter Chapter 17 from Part03.mp3
Extracting chapter Chapter 18 from Part03.mp3
Extracting chapter Chapter 19 from Part04.mp3
Extracting chapter Chapter 20 from Part04.mp3
Extracting chapter Chapter 21 from Part04.mp3
Extracting chapter Chapter 22 from Part04.mp3
Extracting chapter Chapter 23 from Part04.mp3
Extracting chapter Chapter 24 from Part05.mp3
Extracting chapter Chapter 25 from Part05.mp3
Extracting chapter Chapter 26 from Part05.mp3
Extracting chapter Chapter 27 from Part06.mp3
Extracting chapter Chapter 28 from Part06.mp3
Extracting chapter Chapter 29 from Part06.mp3
Extracting chapter Chapter 30 from Part06.mp3
Extracting chapter Chapter 31 from Part07.mp3
Extracting chapter Chapter 32 from Part07.mp3
Extracting chapter Chapter 33 from Part07.mp3
Extracting chapter Chapter 34 from Part07.mp3
Extracting chapter Chapter 35 from Part07.mp3
Extracting chapter Chapter 36 from Part08.mp3
Extracting chapter Chapter 37 from Part08.mp3
Extracting chapter Chapter 38 from Part08.mp3
Extracting chapter Chapter 39 from Part08.mp3
Extracting chapter Chapter 40 from Part09.mp3
Extracting chapter Chapter 41 from Part09.mp3
Extracting chapter Chapter 42 from Part09.mp3
Extracting chapter Chapter 43 from Part09.mp3
Extracting chapter Chapter 44 from Part09.mp3
Extracting chapter Chapter 45 from Part09.mp3
Extracting chapter Chapter 46 from Part09.mp3
Extracting chapter Chapter 47 from Part09.mp3
Extracting chapter Credits from Part09.mp3
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/bin/sonus", line 8, in <module>
    sys.exit(main.main())
             ^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sonus/main.py", line 99, in main
    chapterizer.main(arg, output_path, generic, offset, ffmpeg_debug)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sonus/chapterizer.py", line 226, in main
    merge_chapter_parts(tmpfiles, output_dir, generic_chapters)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/sonus/chapterizer.py", line 153, in merge_chapter_parts
    title = tag_data.get('TALB').text[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'text'
```
my ai told me
`what happened
The `sonus` command failed due to an `AttributeError` in its code. Specifically, it encountered a `NoneType` object when trying to access the `.text` attribute of `tag_data.get('TALB')`.

This error indicates that the `TALB` tag data was not found or correctly parsed in one of the audio files. The processing worked up to the point where it attempted to merge chapter parts, suggesting an issue with the tags of one or more files.

You might want to check the ID3 tags of the files being processed to ensure they contain the necessary `TALB` (Album Title) information. Alternatively, modify the `sonus` code to handle missing tags gracefully. Here's a basic command to start inspecting the ID3 tags of your MP3 files using `ffmpeg`:
for file in *.mp3; do ffmpeg -i "${file}" 2>&1 | grep TALB; done
suggestion`
Its telling me to mess with your code? I just dont feel like i should do that
I normally just move everything over to windows and use inaudible...i just do not like it as much as sonus. the files are names better and it looks better. Plus its way faster (even before i got the m1) Thank you
Heres a link for all the files (im not sure where you prefer them), also i just recalled that windows thing ill go hop on that now

I hope me doing this is helpful
Cerina

**Comment by digitalec** [comment]
I removed your link as it could violate ToS (file sharing).

This issue was fixed in 0.6.5 back on 8/7. Have you updated yet?

https://github.com/digitalec/sonus/releases/tag/v0.6.5

**Comment by cerinawithasea** [comment]
I probably missed the update Thank you, is there a better place to share that information with yoo?
